apis/projectcontour/v1/httpproxy.go

```diff
@@ -616,6 +620,33 @@ type HTTPRequestRedirectPolicy struct {
 	Prefix *string `json:"prefix,omitempty"`
 }
 
+type HTTPInternalRedirectPolicy struct {
+	// MaxInternalRedirects An internal redirect is not handled, unless the number of previous internal
+	// redirects that a downstream request has encountered is lower than this value.
+	// +optional
+	MaxInternalRedirects uint32 `json:"maxInternalRedirects,omitempty"`
+
+	// RedirectResponseCodes If unspecified, only 302 will be treated as internal redirect.
+	// Only 301, 302, 303, 307 and 308 are valid values.
+	// +optional
+	RedirectResponseCodes []uint32 `json:"redirectResponseCodes,omitempty"`
+
+	// AllowCrossSchemeRedirect Allow internal redirect to follow a target URI with a different scheme
+	// than the value of x-forwarded-proto.
+	// SafeOnly allows same scheme redirect and safe cross scheme redirect, which means if the downstream
+	// scheme is HTTPS, both HTTPS and HTTP redirect targets are allowed, but if the downstream scheme
+	// is HTTP, only HTTP redirect targets are allowed.
+	// +kubebuilder:validation:Enum=Always;Never;SafeOnly
```

Comment:
cc @skriss I think we at some point stopped using Enums and defaults via kubebuilder and preferred to do validation in the various processors?

Reply:
We definitely did for ContourConfiguration, but HTTPProxy uses it in a number of places, so I was fine with this.

Reply:
Talking about validation, my understanding was that the annotation where use only to generate the rules in the CRDs, but it still requires a ValidatingWebHook to perform the validation. 

Where is the validation performed in Contour ?